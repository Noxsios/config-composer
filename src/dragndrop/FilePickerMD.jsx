import { useContext, useState } from "react";
import {
  EuiFilePicker,
  // EuiText
} from "@elastic/eui";
import { ToastContainer, toast } from "react-toastify";
import DarkTheme from "@elastic/eui/dist/eui_theme_dark.json";
import MetaContext from "../MetaContext";

const FilePickerMD = () => {
  const [valid, setValid] = useState(true);
  const { meta, setMeta } = useContext(MetaContext);
  const wrongFile = () => toast("Only .md or .markdown files are accepted");

  const onChange = async (files) => {
    if (files.length > 0) {
      const md = await readFileMD(files[0]);
      if (
        files[0].name
          .split(".")
          .slice(-1)[0]
          .match(/md|markdown/g) === null
      ) {
        wrongFile();
        setValid(false);
        setMeta({ ...meta, markdown: "undefined" });
      } else {
        setValid(true);
        setMeta({ ...meta, markdown: md });
      }
    } else {
      setValid(false);
      setMeta({ ...meta, markdown: "undefined" });
    }
  };

  const readFileMD = async (blob) => {
    let textBlob = new Blob([blob], { type: "text" });
    let text = await textBlob.text();
    return text;
  };

  return (
    <div style={{ minWidth: 300, margin: "1rem" }}>
      <EuiFilePicker
        id="md-picker"
        initialPromptText="Markdown"
        onChange={(files) => {
          onChange(files);
        }}
        display="large"
        isInvalid={!valid}
      />
      {/* <EuiText>{meta.markdown !== "undefined" && JSON.stringify(meta.markdown, null, 2)}</EuiText> */}
      <ToastContainer closeOnClick autoClose={5000} toastStyle={toastStyles} closeButton={false} />
    </div>
  );
};

export default FilePickerMD;

const toastStyles = {
  background: DarkTheme.euiCodeBlockBackgroundColor,
};
