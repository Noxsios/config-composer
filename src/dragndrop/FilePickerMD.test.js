import renderer from "react-test-renderer";
import FilePickerMD from "./FilePickerMD";

it("renders history modal skeleton", () => {
  const t = renderer.create(<FilePickerMD />).toJSON();
  expect(t).toMatchSnapshot();
});
