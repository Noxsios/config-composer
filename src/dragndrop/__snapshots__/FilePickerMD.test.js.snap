// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders history modal skeleton 1`] = `
<div
  style={
    Object {
      "margin": "1rem",
      "minWidth": 300,
    }
  }
>
  <div
    className="euiFilePicker euiFilePicker--large"
  >
    <div
      className="euiFilePicker__wrap"
    >
      <input
        aria-describedby="md-picker-filePicker__prompt"
        aria-label="Markdown File Picker"
        className="euiFilePicker__input"
        id="md-picker"
        onChange={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        type="file"
      />
      <div
        className="euiFilePicker__prompt"
        id="md-picker-filePicker__prompt"
      >
        <svg
          aria-hidden={true}
          className="euiIcon euiIcon--large euiIcon-isLoading euiFilePicker__icon"
          focusable="false"
          height={16}
          role="img"
          style={null}
          viewBox="0 0 16 16"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        />
        <div
          className="euiFilePicker__promptText"
        >
          Markdown
        </div>
      </div>
    </div>
  </div>
  <div
    className="Toastify"
  />
</div>
`;
